### ISSUE TITLE
Implement gradient clipping

### ISSUE THREAD

**Issue by JoshVarty** [JoshVarty/HaliteIO#19]
Gradient clipping might help us avoid exploding gradients without having to use such a small learning rate. We should try to implement it. 

[`nn.utils.clip_grad_norm_()`](https://github.com/pytorch/pytorch/blob/27455e9c788a1e95411cd7e0fcbabf6f866ded42/torch/nn/utils/clip_grad.py#L6-L39)
